pacman/model/graphs/machine/machine_graph.py

```diff
@@ -15,20 +15,36 @@
 
 from .machine_vertex import MachineVertex
 from .machine_edge import MachineEdge
+from pacman.model.graphs.graph import Graph
 from pacman.model.graphs import OutgoingEdgePartition
-from pacman.model.graphs import Graph
 
 
 class MachineGraph(Graph):
     """ A graph whose vertices can fit on the chips of a machine.
     """
 
-    __slots__ = []
+    __slots__ = ["_app_graph"]
 
-    def __init__(self, label):
+    def __init__(self, label, application_graph=None):
```

Comment:
Have to be a little bit careful here because of graph filtering (in sPyNNaker right now, but probably should be in FEC now that I've put the filtering rules themselves in the right place); the ordering of operations there is very precise.

Reply:
The reason why I allow `None` is because of the graph front end. I do not wish to change that API.